### PR TITLE
BE structure and migration (user role)

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #2024

### CODE REVIEW

File: src/server/repository/public/user/getMany.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const getMany = async (
       ...Objects.camelize(user),
       roles: roles.map(({ props, ...role }) => ({
         ...Objects.camelize(role),
-        props: { ...Objects.camelize(props), sections: (props as CollaboratorProps).sections },
+        props: { ...Objects.camelize(props), sections: (props as CollaboratorPermissions).sections },
```
Comment (minotogna) [inline src/server/repository/public/user/getMany.ts]: Qui non è permissions ? Invece di props ?

File: src/server/repository/public/user/getOne.ts [hunk 2]
```diff
@@ -58,7 +58,7 @@ export const getOne = async (
           data.roles[0] !== null
             ? data.roles.map(({ props, ...role }) => ({
                 ...Objects.camelize(role),
-                props: { ...Objects.camelize(props), sections: (props as CollaboratorProps).sections },
+                props: { ...Objects.camelize(props), sections: (props as CollaboratorPermissions).sections },
```
Comment (minotogna) [inline src/server/repository/public/user/getOne.ts]: Anche qui non è permissions ?
Io aggiungerei uno userrole adapter (vedi server/repository/adapters)

File: src/server/repository/public/userRole/updateSectionAuth.ts [hunk 3]
```diff
@@ -1,30 +1,33 @@
...
         returning *;
     `,
       [JSON.stringify(sections), id]
     )
     .then((data) => ({
       ...Objects.camelize(data),
-      props: { ...Objects.camelize(props), sections: (props as CollaboratorProps).sections },
+      permissions: {
+        ...Objects.camelize(data.permissions),
+        sections: (data.permissions as CollaboratorPermissions).sections,
```
Comment (minotogna) [inline src/server/repository/public/userRole/updateSectionAuth.ts]: Qui anche va usato l'adapter. L'adapter deve gestire sia props che permissions

File: src/meta/api/endpoint/ApiEndPoint.ts [hunk 4]
```diff
@@ -95,6 +95,7 @@ export const ApiEndPoint = {
     roles: () => apiPath('users', 'roles'),
     adminRole: () => apiPath('users', 'admin-role'),
     sectionAuth: () => apiPath('users', 'section-auth'),
+    roleProps: () => apiPath('users', 'role-props'),
```
Comment (minotogna) [inline src/meta/api/endpoint/ApiEndPoint.ts]: Userei /users/role/props

Comment (rfontanarosa) [reply src/meta/api/endpoint/ApiEndPoint.ts]: ok. I would like to discuss about section-auth, probably we should change that too

File: src/client/store/ui/userManagement/actions/updateSectionAuth.ts [hunk 5]
```diff
@@ -3,13 +3,13 @@ import axios from 'axios'
...
-import { CollaboratorProps, CollaboratorSectionsProp, RoleName, UserRole } from '@meta/user'
+import { Collaborator, CollaboratorSectionsPermission } from '@meta/user'
 
 export const updateSectionAuth = createAsyncThunk<
-  UserRole<RoleName, CollaboratorProps>,
+  Collaborator,
   {
     id: number
-    sections: CollaboratorSectionsProp
+    sections: CollaboratorSectionsPermission
```
Comment (sorja) [inline src/client/store/ui/userManagement/actions/updateSectionAuth.ts:12]: can we rename sections to sectionsPermission or something?
A bit misleading maybe

File: src/meta/user/userRole.ts [hunk 6]
```diff
@@ -9,37 +9,86 @@ export enum RoleName {
...
+    street?: string
+    zipCode?: string
+    poBox?: string
+    city?: string
+    country?: CountryIso
+  }
+  primaryEmail?: string
+  secondaryEmail?: string
+  primaryPhoneNo?: string
+  secondaryPhoneNo?: string
```
Comment (sorja) [inline src/meta/user/userRole.ts]: can we call it {primary|secondary}PhoneNumber?

File: src/server/repository/public/user/getOne.ts [hunk 7]
```diff
@@ -54,13 +55,7 @@ export const getOne = async (
...
       return {
         ...Objects.camelize(data),
-        roles:
-          data.roles[0] !== null
-            ? data.roles.map(({ props, ...role }) => ({
-                ...Objects.camelize(role),
-                props: { ...Objects.camelize(props), sections: (props as CollaboratorProps).sections },
-              }))
-            : [],
+        roles: data.roles[0] !== null ? data.roles.map(UserRoleAdapter) : [],
```
Comment (sorja) [inline src/server/repository/public/user/getOne.ts:58]: Could the syntax be `roles: data.roles.map(UserRoleAdapter)` and null case handled by adapter?

Comment (sorja) [inline src/server/repository/public/user/getOne.ts]: would `data.roles?.map() ?? []` work?

Comment (minotogna) [reply src/server/repository/public/user/getOne.ts]: or even (data.roles ?? []).map(UserRoleAdapter)

File: src/server/repository/public/user/getMany.ts [hunk 8]
```diff
@@ -2,9 +2,10 @@ import { Objects } from '@utils/objects'
 
 import { CountryIso } from '@meta/area'
 import { Assessment, Cycle } from '@meta/assessment'
-import { CollaboratorProps, RoleName, User } from '@meta/user'
+import { RoleName, User } from '@meta/user'
 
 import { BaseProtocol, DB } from '@server/db'
+import { UserRoleAdapter } from '@server/repository/adapter'
 
 import { fields } from './fields'
```
Comment (minotogna) [inline src/server/repository/public/user/getMany.ts:10]: check if fields must be updated

Comment (minotogna) [reply src/server/repository/public/user/getMany.ts:10]: I meant user role props, if permissions are included and props are included conditionally   in select

Comment (rfontanarosa) [reply src/server/repository/public/user/getMany.ts:10]: excluded from query

`jsonb_agg(to_jsonb(ur.* ) - 'props')`

**Review by sorja — APPROVED** [review]

**Pull request merged**